src/promql/src/planner.rs

```diff
@@ -79,10 +82,55 @@ impl<S: ContextProvider> PromPlanner<S> {
 
     pub fn prom_expr_to_plan(&mut self, prom_expr: PromExpr) -> Result<LogicalPlan> {
         let res = match &prom_expr {
-            PromExpr::AggregateExpr { .. } => UnsupportedExprSnafu {
-                name: "Prom Aggregate",
+            PromExpr::AggregateExpr {
+                op,
+                expr,
+                // TODO(ruihang): support param
+                param: _param,
+                grouping,
+                without,
+            } => {
+                let input = self.prom_expr_to_plan(*expr.clone())?;
+
+                // calculate columns to group by
+                let schema = input.schema();
+                let group_columns_indices = grouping
+                    .iter()
+                    .map(|label| {
+                        schema
+                            .index_of_column_by_name(None, label)
+                            .with_context(|_| LabelNotFoundSnafu {
+                                table: self.ctx.table_name.clone().unwrap(),
+                            })
+                    })
+                    .collect::<Result<HashSet<_>>>()?;
```

Comment:
So we don't keep the group fields in the order of `grouping`, I think it may change the result.

Reply:
I take a look at prometheus's return json, the tags (labels) are placed inside a map. So I think it's good to to do like this here. And for `group without` there is not ordering in the begining.
```json
            {
                "metric": {
                    "__name__": "container_cpu_system_seconds_total",
                    "id": "/system.slice/ModemManager.service",
                    "instance": "cadvisor:8080",
                    "job": "cadvisor"
                },
                "value": [
                    1673582966.700,
                    "0.07"
                ]
            },
            {
                "metric": {
                    "__name__": "container_cpu_system_seconds_total",
                    "id": "/system.slice/accounts-daemon.service",
                    "instance": "cadvisor:8080",
                    "job": "cadvisor"
                },
                "value": [
                    1673582966.700,
                    "47.16"
                ]
            },
```